x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/BoostedTreeParams.java

```diff
@@ -214,6 +224,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalDouble(softTreeDepthTolerance);
         out.writeOptionalDouble(downsampleFactor);
         out.writeOptionalVInt(maxOptimizationRoundsPerHyperparameter);
+        out.writeOptionalBoolean(earlyStoppingEnabled);
```

Comment:
This needs to be protected by version guards, for example:

```
if (out.getVersion().onOrAfter(V_7_12_0)) {
    out.writeOptionalBoolean(earlyStoppingEnabled);
}
```

Whether you can get away with `V_7_12_0` from the start or whether you have to use `V_8_0_0` in the initial master branch PR, then change to `V_7_12_0` after backporting to 7.x depends on how good our BWC coverage is: if we have good coverage you'll need to start off with `V_8_0_0` and if we don't you can go straight to `V_7_12_0`.

Reply:
Fixed :+1: